client/cherami/connection.go

```diff
@@ -300,8 +344,16 @@ func (conn *connection) isOpened() bool {
 	return atomic.LoadInt32(&conn.opened) != 0
 }
 
+// isClosed should return true if either closed it set
+// or if "drain" is set, which means the connection has already
+// stopped the write pump.
+// This is needed to make sure if reconfigure gives the same
+// inputhost, we should open up a new connection object (most
+// likely for a new extent). In the worst case, if for some reason
+// the controller returns the same old extent which is draining,
+// the server will anyway reject the connection
 func (conn *connection) isClosed() bool {
-	return atomic.LoadInt32(&conn.closed) != 0
+	return (atomic.LoadInt32(&conn.closed) != 0 || atomic.LoadInt32(&conn.drained) != 0)
```

Comment:
channels are thread safe.. not sure about the race.. Will leave it up to you resolve it.